Ponca/src/Fitting/basket.h

```diff
@@ -237,7 +261,12 @@ namespace internal
         }
     }; // class Basket
 
-#undef BASKET_TP
+    template <class P, class W, int Type,
+        template <class, class, typename> class Ext0,
+        template <class, class, typename> class... Exts>
+    class BasketAutoDiff : public internal::BasketAutoDiffAggregate<P, W, Type, Ext0, Exts...>::type
```

Comment:
Is the `AutoDiff` feature in draft or can we review it?
Maybe we can move the `AutoDiff` feature in another PR?

Reply:
This was added as a followup to #106. The aggregate part should be ready for review, but I don't know what the functionality of the class is going to be yet, so I can remove this from this PR for now, if necessary.

Reply:
@azaleostu could you remove the autodiff version so we can merge this refactoring, and then add autodiff in another PR (with appropriate testing)